### PR TITLE
Add partial support for Pylint.

### PR DESCRIPTION
**Pull request by rjeffman**
This PR adds support for running Pylint 2.8.2, over the `plugins` directory, as a pre-commit hook, and as a Github workflow action  to be executed or pushes and pull requests.

The Github workflow run pylint against all python files under the `plugins` directory.

The pre-commit hook runs pylint only in the modified Python files, to speed up the commit process. This is independent of directory, in the hope that all files will be either fixed or have the explicit checks disabled over time. 

All Pylint warnings for the `plugins` directory were either fixed or disabled, based, mostly, on code readability. The `roles` directory is not part of this PR, and should be addressed later.

### CODE REVIEW

**Comment by rjeffman** [comment]
This PR should not be merged before CI is working again, due to changes, although simple, made in many files.

File: plugins/module_utils/ansible_freeipa_module.py [hunk 1]
```diff
@@ -489,8 +480,7 @@ def __init__(self, ansible_module):
 
         def __getitem__(self, key):
             param = self.mapping[key]
-            if param is not None:
-                return _afm_convert(param)
+            return _afm_convert(param) if param is not None else None
```
Comment (t-woerner) [inline plugins/module_utils/ansible_freeipa_module.py:483]: Please change it, it is hard to read. What about `return None if param is None else _afm_convert(param)`

File: plugins/modules/ipaservice.py [hunk 2]
```diff
@@ -839,7 +839,12 @@ def main():
                         changed = True
                 else:
                     changed = True
+            except ipalib.errors.EmptyModlist:
```
Comment (t-woerner) [inline plugins/modules/ipaservice.py:842]: Why have you added these lines?

File: plugins/modules/ipauser.py [hunk 3]
```diff
@@ -599,17 +598,14 @@ def gen_args(first, last, fullname, displayname, initials, homedir, shell,
...
-                     shell, email, principal, principalexpiration,
-                     passwordexpiration, password, random, uid, gid, city,
-                     phone, mobile, pager, fax, orgunit, title, manager,
-                     carlicense, sshpubkey, userauthtype, userclass, radius,
-                     radiususer, departmentnumber, employeenumber,
-                     employeetype, preferredlanguage, certificate,
-                     certmapdata, noprivate, nomembers, preserve,
-                     update_password):
-
+def check_parameters(  # pylint: disable=unused-argument
```
Comment (t-woerner) [inline plugins/modules/ipauser.py:601]: Wouldn't it be better to remove the unused args?

File: plugins/modules/ipavault.py [hunk 4]
```diff
@@ -443,12 +444,12 @@ def data_storage_args(vault_type, args, data, password, password_file,
     return _args
 
 
-def check_parameters(module, state, action, description, username, service,
-                     shared, users, groups, services, owners, ownergroups,
-                     ownerservices, vault_type, salt, password, password_file,
-                     public_key, public_key_file, private_key,
-                     private_key_file, vault_data, datafile_in, datafile_out,
-                     new_password, new_password_file):
+def check_parameters(  # pylint: disable=unused-argument
```
Comment (t-woerner) [inline plugins/modules/ipavault.py:447]: Also here.

File: plugins/modules/ipavault.py [hunk 5]
```diff
@@ -491,11 +492,11 @@ def check_parameters(module, state, action, description, username, service,
                     "action '%s'" % (arg, state, action))
 
 
-def check_encryption_params(module, state, action, vault_type, salt,
-                            password, password_file, public_key,
-                            public_key_file, private_key, private_key_file,
-                            vault_data, datafile_in, datafile_out,
-                            new_password, new_password_file, res_find):
+def check_encryption_params(  # pylint: disable=unused-argument
```
Comment (t-woerner) [inline plugins/modules/ipavault.py:495]: And here

File: tests/service/certificate/test_service_certificate.yml [hunk 6]
```diff
@@ -76,7 +76,7 @@
       ok_as_delegate: no
       ok_to_auth_as_delegate: no
     register: result
-    failed_when: not result.changed
+    failed_when: not result.changed and result.failed
```
Comment (t-woerner) [inline tests/service/certificate/test_service_certificate.yml:79]: This change is unrelated.

File: tests/service/env_cleanup.yml [hunk 7]
```diff
@@ -8,7 +8,9 @@
       - "HTTP/{{ nohost_fqdn }}"
       - HTTP/svc.ihavenodns.info
       - HTTP/no.idontexist.local
+      - HTTP/no.idontexist.info
```
Comment (t-woerner) [inline tests/service/env_cleanup.yml:11]: This change is unrelated.

File: tests/service/env_setup.yml [hunk 8]
```diff
@@ -61,13 +61,3 @@
   ipahostgroup:
     ipaadmin_password: SomeADMINpassword
     name: hostgroup02
```
Comment (t-woerner) [inline tests/service/env_setup.yml:63]: This change is unrelated.

File: tests/service/test_service.yml [hunk 9]
```diff
@@ -39,14 +39,14 @@
           ok_as_delegate: no
           ok_to_auth_as_delegate: no
         register: result
-        failed_when: not result.changed
+        failed_when: not result.changed or result.failed
```
Comment (t-woerner) [inline tests/service/test_service.yml:42]: This change is unrelated.

File: tests/service/test_service_disable.yml [hunk 10]
```diff
@@ -40,7 +40,7 @@
       - "{{ lookup('file', 'cert1.b64', rstrip=False) }}"
       force: no
     register: result
-    failed_when: not result.changed
+    failed_when: not result.changed or result.failed
```
Comment (t-woerner) [inline tests/service/test_service_disable.yml:43]: This change is unrelated.

File: tests/service/test_service_keytab.yml [hunk 11]
```diff
@@ -33,7 +33,7 @@
       - user02
       action: member
     register: result
-    failed_when: not result.changed
+    failed_when: not result.changed or result.failed
```
Comment (t-woerner) [inline tests/service/test_service_keytab.yml:36]: This change is unrelated.

File: tests/service/test_service_without_skip_host_check.yml [hunk 12]
```diff
@@ -24,22 +24,22 @@
       ok_as_delegate: no
       ok_to_auth_as_delegate: no
     register: result
-    failed_when: not result.changed
+    failed_when: not result.changed or result.failed
```
Comment (t-woerner) [inline tests/service/test_service_without_skip_host_check.yml:27]: This change is unrelated.

Comment (rjeffman) [reply tests/service/test_service_without_skip_host_check.yml:27]: This change is in a TEMP commit, so service tests pass.

File: tests/service/test_service_disable.yml (hunk 10 above)
Comment (rjeffman) [reply tests/service/test_service_disable.yml:43]: This change is in a TEMP commit, so service tests pass.

File: tests/service/test_service_keytab.yml (hunk 11 above)
Comment (rjeffman) [reply tests/service/test_service_keytab.yml:36]: This change is in a TEMP commit, so service tests pass.

File: tests/service/env_setup.yml (hunk 8 above)
Comment (rjeffman) [reply tests/service/env_setup.yml:63]: This change is in a TEMP commit, so service tests pass.

File: tests/service/env_cleanup.yml (hunk 7 above)
Comment (rjeffman) [reply tests/service/env_cleanup.yml:11]: This change is in a TEMP commit, so service tests pass.

File: tests/service/certificate/test_service_certificate.yml (hunk 6 above)
Comment (rjeffman) [reply tests/service/certificate/test_service_certificate.yml:79]: This change is in a TEMP commit, so service tests pass.

File: plugins/modules/ipaservice.py (hunk 2 above)
Comment (rjeffman) [reply plugins/modules/ipaservice.py:842]: This change is in a TEMP commit, so service tests pass.

File: plugins/modules/ipauser.py (hunk 3 above)
Comment (rjeffman) [reply plugins/modules/ipauser.py:601]: Not in this case. As we use `vars()` to evaluate parameters, they need to be locally accessible, therefore, we need the parameters (and they will actually be used).

File: plugins/modules/ipavault.py (hunk 4 above)
Comment (rjeffman) [reply plugins/modules/ipavault.py:447]: We need them it in `check_parameter`-like functions.

File: plugins/modules/ipavault.py (hunk 5 above)
Comment (rjeffman) [reply plugins/modules/ipavault.py:495]: We need them it in `check_parameter`-like functions.

**Review by t-woerner — APPROVED** [review]
LGTM
